micro-rdk/src/common/conn/viam.rs

```diff
@@ -640,13 +667,16 @@ where
             webrtc_config: &self.webrtc_configuration,
             network,
             incomming_connection_manager: IncomingConnectionManager::new(
-                self.max_concurrent_connections as usize,
+                self.max_concurrent_connections,
             ),
             robot_config: &config,
-            local_signaling_server: Arc::new(SignalingServer::new(
+            #[cfg(feature = "local-signaling")]
+            local_signaling_server: Some(Arc::new(SignalingServer::new(
```

Comment:
@mattjperez - Here is where we should decide to not instantiate the `SignalingServer` if the max conns is too low.